### ISSUE TITLE
Blocking reverse lookups for specific ips

### ISSUE THREAD

**Issue by Code-Slave** [TechnitiumSoftware/DnsServer#369]
Havent figured if there is a way to do this yet. So linksys mesh routers have a big bug for over a year that spams reverse lookups of local ips. (x.x.x.x.in.addr.arpa) is there a way i can block these? even though the ips have dns records it still gets spammed (25-30k requests in a day) so id like to block them from 3 specific ips. Thanks

**Comment by ShreyasZare** [comment]
Yes, you can do that using the Drop Requests DNS App. Edit the app's config which is in json format and add 3 entries inside `blockedQuestions` array. The requests will then be dropped before reaching the DNS server core so you wont see it in any logs or stats.

**Comment by Code-Slave** [comment]
awesome. thanks

**Comment by ShreyasZare** [comment]
> awesome. thanks. so add those ips in blocked questions?

There are two ways to use the app: either block the IP using `blockedNetworks` array so that all queries from that IP are blocked, OR you use the `blockedQuestions` so that any query for that specific domain name is dropped irrespective of the request IP address.

Using the `blockedQuestions` to block the domain name is good. If you block the IP address then that device wont be able to resolve any other domain name.

**Comment by Code-Slave** [comment]
perfect. thanks

**Comment by Code-Slave** [comment]
sorry for the addon. 

So it looks like dhcp creates a reverse lookup zone. called
1.168.192.in-addr.arpa

the only entries in that are dhcp clients. Im wondering if i need to add my other ips to that as that seems to be contributing as it cant find the reverse lookup for specific ips

**Comment by ShreyasZare** [comment]
> So it looks like dhcp creates a reverse lookup zone. called 1.168.192.in-addr.arpa
> 
> the only entries in that are dhcp clients. Im wondering if i need to add my other ips to that as that seems to be contributing as it cant find the reverse lookup for specific ips

That reverse zone will be populated automatically by the DHCP server when it allocated IP address. If you have any device with statically configured IP then you can add a manual entry in that zone. For example, your router's IP will be static so you can add a PTR record for it.

**Comment by ShreyasZare** [comment]
This since by default the config also contains `allowedNetworks` with all private IP ranges defined. The allowed networks is checked first and thus your blocked network config is not working. You need to remove the "192.168.0.0/16" from allowed networks.

**Comment by ShreyasZare** [comment]
I will update the app to fix the check order so that it checks the blocked network first.

**Comment by ShreyasZare** [comment]
Yes, just remove it from allowed networks and it will work.

**Comment by Code-Slave** [comment]
all good. thanks
